src/scratch_impl.h

```diff
@@ -19,22 +19,32 @@ static secp256k1_scratch* secp256k1_scratch_create(const secp256k1_callback* err
     secp256k1_scratch* ret = (secp256k1_scratch*)checked_malloc(error_callback, sizeof(*ret));
```

Comment:
Can we unify this malloc with the other one too? Also, `scratch_crate` kind of promises to allocate at most `max_size` bytes but we allocate more.

Reply:
I guess we should rename `max_size` ... the question is whether it will be more annoying to the user if (a) the function allocates more than they tell it to; (b) the function allocates exactly what they say, but gives them less scratch space than they intended.

My guess is that (a) is the better way to go here, but we should document this.

As for unifying the `malloc` calls, yeah, sure, I can do that.